python_modules/libraries/dagster-components/dagster_components/core/component_defs_builder.py

```diff
@@ -116,3 +122,27 @@ def load_defs(
         )
         all_defs.append(defs)
     return Definitions.merge(*all_defs)
+
+
+def get_passed_assets_defs(defs: Definitions) -> Sequence[AssetsDefinition]:
```

Comment:
`get_all_passed_assets_as_assets_defs` ?

I know the name is long but trying to avoid confusion. Unfortunately the behavior is not intuitive and I think we need a shitty name to accurately describe what is happening